starlette/datastructures.py

```diff
@@ -414,28 +413,25 @@ class UploadFile:
     An uploaded file included as part of the request data.
     """
 
-    spool_max_size = 1024 * 1024
     file: typing.BinaryIO
     headers: "Headers"
 
     def __init__(
         self,
         filename: str,
-        file: typing.Optional[typing.BinaryIO] = None,
+        file: typing.BinaryIO,
```

Comment:
We could also make this strictly `file: SpooledTemporaryFile`

Reply:
Typing-wise, isn't there a better type for this situation? Also... It's not an implementation detail, it's written on our docs that the file is a `SpooledTemporaryFile`.

Reply:
Right now there's three options:
1. Use `IO[bytes]`
2. Use `SpooledTemporaryFile` directly
3. Write a protocol for just the methods we need

Source: https://github.com/python/typing/discussions/829#discussioncomment-1150451

I don't think we should choose (2) and I think we should change our docs. Let's not lock ourselves into concrete implementations if we don't need to.

I think (1) has issues (as mentioned in that discussion and others) but is widely used so it's pretty defensible if a user had to add a `# type: ignore` to that parameter if they pass in some less common file-like object.

(3) is the _correct_ option but also the most boilerplate / overhead for us. It would look like:

```python
class FileLike(Protocol):
    def write(self, __data: bytes) -> None:  ...
    def read(self, __n: int) -> bytes:  ...
    def seek(self, __pos: int) -> None:  ...
    def close(self) -> None:  ...
```

Reply:
> I think we should change our docs

That's enough for me. :+1: